javascript/ql/src/semmle/javascript/frameworks/NodeJSLib.qll

```diff
@@ -506,4 +506,295 @@ module NodeJSLib {
     }
   }
 
+  /**
+   * A data flow node that is a Node.js HTTP or HTTPS client request.
+   */
+  abstract class ClientRequest extends DataFlow::TrackedNode {
```

Comment:
This seems to overlap with the existing class `RequestExpr`. What's the rationale for introducing this class, and why is it a `TrackedNode`? Have you seen cases where we need to track requests inter-procedurally?

Reply:
`RequestExpr` is used in a NodeJS server handling an incoming request, while `ClientRequest` is when the NodeJS or Electron libraries are used to send a request, as in the call at line 18 of javascript/ql/test/library-tests/frameworks/NodeJSLib/src/http.js

I think I added that because I saw a case where we need to track them interprocedurally - I'll go back through the projects I was testing on and see if I actually needed it.

Reply:
It looks like I added it in an attempt to handle a use of `bind` to create the request callback. It didn't work, so I'll remove it and see if there's a better way to handle `bind`.

Reply:
OK, that's not clear from the description or the name. Please consider whether there is a better name, and extend the qldoc with an example.

`bind` is tricky; I'd say let's ignore it for now.